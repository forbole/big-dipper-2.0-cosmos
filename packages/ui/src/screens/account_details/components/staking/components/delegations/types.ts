import type { DelegationType } from '@/screens/account_details/components/staking/types';

export type ItemType = DelegationType;
