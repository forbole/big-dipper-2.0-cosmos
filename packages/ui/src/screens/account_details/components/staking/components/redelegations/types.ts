import type { RedelegationType } from '@/screens/account_details/components/staking/types';

export type ItemType = RedelegationType;
