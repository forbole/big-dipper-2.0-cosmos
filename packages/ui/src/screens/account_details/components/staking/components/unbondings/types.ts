import type { UnbondingType } from '@/screens/account_details/components/staking/types';

export type ItemType = UnbondingType;
