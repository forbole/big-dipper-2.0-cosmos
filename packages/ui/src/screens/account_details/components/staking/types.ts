export type StakingType<g> = {
  data: {
    [value: number]: g[];
  };
  count: number;
  loading: boolean;
};

export interface DelegationType {
  validator: string;
  amount: TokenUnit;
  reward: TokenUnit;
  commission?: number;
}

export interface RedelegationType {
  from: string;
  to: string;
  amount: TokenUnit;
  completionTime: string;
}

export interface UnbondingType {
  validator: string;
  amount: TokenUnit;
  completionTime: string;
}

export type DelegationsType = StakingType<DelegationType>;
export type RedelegationsType = StakingType<RedelegationType>;
export type UnbondingsType = StakingType<UnbondingType>;

export interface StakingState {
  tab: number;
  delegations: DelegationsType;
  redelegations: RedelegationsType;
  unbondings: UnbondingsType;
}
