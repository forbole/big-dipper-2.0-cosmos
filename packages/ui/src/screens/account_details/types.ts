export interface OverviewType {
  address: string;
  withdrawalAddress: string;
}

export interface BalanceType {
  available: TokenUnit;
  delegate: TokenUnit;
  unbonding: TokenUnit;
  reward: TokenUnit;
  commission?: TokenUnit;
  total: TokenUnit;
}

export interface OtherTokenType {
  denom: string;
  available: TokenUnit;
  reward: TokenUnit;
  commission: TokenUnit;
}

export interface RewardsType {
  [value: string]: TokenUnit;
}

export interface AccountDetailState {
  loading: boolean;
  exists: boolean;
  desmosProfile: DesmosProfile | null;
  overview: OverviewType;
  balance: BalanceType;
  otherTokens: {
    data: OtherTokenType[];
    count: number;
  };
  rewards: RewardsType;
}
