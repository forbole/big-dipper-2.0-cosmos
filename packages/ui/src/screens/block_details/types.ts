export type OverviewType = {
  height: number;
  hash: string;
  txs: number;
  timestamp: string;
  proposer: string;
  // votingPower: number;
};
