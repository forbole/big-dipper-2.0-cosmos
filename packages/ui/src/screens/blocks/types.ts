export interface BlockType {
  height: number;
  txs: number;
  timestamp: string;
  proposer: string;
  hash: string;
}

export interface BlocksState {
  loading: boolean;
  exists: boolean;
  hasNextPage: boolean;
  isNextPageLoading: boolean;
  items: BlockType[];
}

export type ItemType = BlockType;
