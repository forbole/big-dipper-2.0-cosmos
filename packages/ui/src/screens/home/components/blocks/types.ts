export interface BlockType {
  height: number;
  txs: number;
  timestamp: string;
  proposer: string;
  hash: string;
}

export interface BlocksState {
  loading: boolean;
  items: BlockType[];
}

export type ItemType = BlockType;
