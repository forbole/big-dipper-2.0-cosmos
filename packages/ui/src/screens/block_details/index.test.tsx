import { ApolloClient, ApolloProvider, from, InMemoryCache } from '@apollo/client';
import { MockedProvider } from '@apollo/client/testing';
import renderer from 'react-test-renderer';
import { BlockDetailsDocument } from '@/graphql/types/general_types';
import BlockDetails from '@/screens/block_details';
import MockTheme from '@/tests/mocks/MockTheme';
import wait from '@/tests/utils/wait';

// ==================================
// mocks
// ==================================
jest.mock('next/router', () => ({
  ...jest.requireActual('next/router'),
  useRouter: () => ({
    query: {
      height: 300,
    },
  }),
}));
jest.mock('@/components/layout', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="Layout" {...props} />
));
jest.mock('@/components/load_and_exist', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="LoadAndExist" {...props} />
));

jest.mock(
  '@/screens/block_details/components/overview',
  () => (props: JSX.IntrinsicElements['div']) => <div id="Overview" {...props} />
);
jest.mock(
  '@/screens/block_details/components/transactions',
  () => (props: JSX.IntrinsicElements['div']) => <div id="Transactions" {...props} />
);
jest.mock(
  '@/screens/block_details/components/signatures',
  () => (props: JSX.IntrinsicElements['div']) => <div id="Signatures" {...props} />
);

const mockAverageBlockTime = jest.fn().mockReturnValue({
  data: {
    transaction: [
      {
        height: 999,
        hash: '393310C681CB39E09CD3AC16C600DBFDACB2DF5085277DA81E52698620C06136',
        logs: [],
        messages: [
          {
            '@type': '/cosmos.staking.v1beta1.MsgDelegate',
            amount: {
              denom: 'udaric',
              amount: '27466368',
            },
            delegator_address: 'desmos18kvwy5hzcu3ss08lcfcnx0eajuecg69ujmkwjr',
            validator_address: 'desmosvaloper18kvwy5hzcu3ss08lcfcnx0eajuecg69uvk76c3',
          },
        ],
        success: true,
      },
    ],
    block: [
      {
        height: 999,
        hash: 'E568ACFE5717F79A44979563410FF7F6C3043A07307E541EF21E15FC478C3DF0',
        timestamp: '2021-04-27T16:27:34.331769',
        txs: 1,
        validator: {
          validatorInfo: {
            operatorAddress: 'desmosvaloper18kvwy5hzcu3ss08lcfcnx0eajuecg69uvk76c3',
          },
        },
      },
    ],
    preCommitsAggregate: {
      aggregate: {
        sum: {
          votingPower: 7304,
        },
      },
    },
    preCommits: [
      {
        validator: {
          validatorInfo: {
            operatorAddress: 'desmosvaloper1qlh47ty9ah2d5e0xq6gsvqjvfulljl9602k7f9',
          },
        },
      },
    ],
  },
});

// ==================================
// unit tests
// ==================================
describe('screen: BlockDetails', () => {
  it('matches snapshot', async () => {
    let component: renderer.ReactTestRenderer | undefined;

    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={new ApolloClient({ link: from([]), cache: new InMemoryCache() })}>
          <MockedProvider
            mocks={[
              {
                request: {
                  query: BlockDetailsDocument,
                  variables: { height: 300, signatureHeight: 301 },
                },
                result: mockAverageBlockTime,
              },
            ]}
          >
            <MockTheme>
              <BlockDetails />
            </MockTheme>
          </MockedProvider>
        </ApolloProvider>
      );
    });
    await wait(renderer.act);

    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
