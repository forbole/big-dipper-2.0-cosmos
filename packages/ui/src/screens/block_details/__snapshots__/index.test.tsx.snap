// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: BlockDetails matches snapshot 1`] = `
.emotion-0 {
  padding: 16px;
  display: grid;
  grid-template-rows: auto auto 1fr;
  grid-template-columns: 1fr;
  gap: 8px;
}

@media (min-width: 1280px) {
  .emotion-0 {
    padding: 16px 24px;
  }
}

.emotion-0 a {
  color: #4092CD;
}

@media (min-width:1280px) {
  .emotion-0 {
    gap: 16px;
  }
}

.emotion-1 {
  height: 450px;
}

@media (min-width:1280px) {
  .emotion-1 {
    height: 450px;
  }
}

<div
  id="Layout"
  navTitle="blockDetails"
>
  <div
    exists={true}
    id="LoadAndExist"
    loading={false}
  >
    <span
      className="emotion-0"
    >
      <div
        hash="E568ACFE5717F79A44979563410FF7F6C3043A07307E541EF21E15FC478C3DF0"
        height={999}
        id="Overview"
        proposer="desmosvaloper18kvwy5hzcu3ss08lcfcnx0eajuecg69uvk76c3"
        timestamp="2021-04-27T16:27:34.331769"
        txs={1}
      />
      <div
        className="emotion-1"
        id="Signatures"
        signatures={
          [
            "desmosvaloper1qlh47ty9ah2d5e0xq6gsvqjvfulljl9602k7f9",
          ]
        }
      />
      <div
        id="Transactions"
        transactions={
          [
            {
              "hash": "393310C681CB39E09CD3AC16C600DBFDACB2DF5085277DA81E52698620C06136",
              "height": 999,
              "messages": {
                "count": 1,
                "items": [
                  {
                    "amount": {
                      "amount": "27466368",
                      "denom": "udaric",
                    },
                    "category": "staking",
                    "delegatorAddress": "desmos18kvwy5hzcu3ss08lcfcnx0eajuecg69ujmkwjr",
                    "json": {
                      "@type": "/cosmos.staking.v1beta1.MsgDelegate",
                      "amount": {
                        "amount": "27466368",
                        "denom": "udaric",
                      },
                      "delegator_address": "desmos18kvwy5hzcu3ss08lcfcnx0eajuecg69ujmkwjr",
                      "validator_address": "desmosvaloper18kvwy5hzcu3ss08lcfcnx0eajuecg69uvk76c3",
                    },
                    "type": "/cosmos.staking.v1beta1.MsgDelegate",
                    "validatorAddress": "desmosvaloper18kvwy5hzcu3ss08lcfcnx0eajuecg69uvk76c3",
                  },
                ],
              },
              "success": true,
              "timestamp": "2021-04-27T16:27:34.331769",
              "type": [
                "Delegate",
              ],
            },
          ]
        }
      />
    </span>
  </div>
</div>
`;
